README.md

```diff
@@ -62,6 +62,47 @@ npx nx build (app_name)
 
 The build artifacts will be stored in the `dist/` directory. Note: this always produces a production build.
 
+### A word on authentication
+
+GeoNetwork-UI applications rely on the GeoNetwork authentication mechanism. This means that if the user is authenticated in GeoNetwork, they will have access to authenticated features in the corresponding GeoNetwork-UI apps.
+
+There are a few caveats, depending on the deployment scenario:
+
+1. GeoNetwork and GeoNetwork-UI are deployed on the same host, e.g. https://my.host/geonetwork and https://my.host/datahub
+   * In this scenario, requests from the GeoNetwork-UI app to GeoNetwork are *not* [cross-origin requests](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS#what_requests_use_cors), so CORS rules do not apply
+   * GeoNetwork has an XSRF protection by default, which *will* make authenticated requests fail unless the following is done:
+     * either make sure that the XSRF cookies sent by GeoNetwork have a `path` value of `/`; this is typically done like so in GeoNetwork:
+       ```diff
+       --- a/web/src/main/webapp/WEB-INF/config-security/config-security-core.xml
+       +++ b/web/src/main/webapp/WEB-INF/config-security/config-security-core.xml
+       @@ -361,6 +361,7 @@
+          <bean class="org.fao.geonet.security.web.csrf.CookieCsrfTokenRepository"
+                id="csrfTokenRepository">
+            <property name="cookieHttpOnly" value="false"/>
+       +    <property name="cookiePath" value="/"/>
+          </bean>
+       ```
+
+     * or disable the XSRF protection selectively for non-critical endpoints of GeoNetwork, e.g. https://my.host/geonetwork/srv/api/userSelections for marking records as favorites; this is typically done like so in GeoNetwork:
+       ```diff
+       --- a/web/src/main/webapp/WEB-INF/config-security/config-security-core.xml
+       +++ b/web/src/main/webapp/WEB-INF/config-security/config-security-core.xml
+       @@ -374,6 +374,9 @@
+                <value>/[a-zA-Z0-9_\-]+/[a-z]{2,3}/csw!?.*</value>
+                <value>/[a-zA-Z0-9_\-]+/api/search/.*</value>
+                <value>/[a-zA-Z0-9_\-]+/api/site</value>
+       +        <value>/[a-zA-Z0-9_\-]+/api/userselections.*</value>
+              </set>
+            </constructor-arg>
+          </bean>
+       ```
+
+2. GeoNetwork and GeoNetwork-UI are *not* deployed on the same host, e.g. https://my.host/geonetwork and https://another.org/datahub
+   * In this scenario, even if CORS settings are correctly set up on GeoNetwork side, most authenticated request will probably fail because by default they are not sent with the [`withCredentials: true`](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/withCredentials) option
```

Comment:
Does it mean we need to add interceptors to add this header ?

Reply:
No, we just have to add this option in the API client configuration. Currently it is done only in the datafeeder here:

https://github.com/geonetwork/geonetwork-ui/blob/2a695f52614ab26dd36a2a8ebe4c48e621c5088e/apps/datafeeder/src/app/app.module.ts#L43

Maybe we could do it with a setting, or maybe we can enable it for everyone, I'm not sure yet

Reply:
I think we could offer a setting like "allow authenticated cross origin requests"; then it would mean that GN would have to be configured to not send `*` in the `Access-Control-Allow-Origin` header, but the actual origin instead, e.g. `Access-Control-Allow-Origin: my-other-site.org`.

This is an optional behaviour that requires tweaking the backend more; the default behaviour would be to not do authenticated requests in a cross origin scenario (e.g. only public search).